### ISSUE TITLE
Enabling doctests

### ISSUE THREAD

**Issue by sobolevn** [bobthemighty/punq#13]
I am trying to enable `doctests` with `pytest --doctest-modules`, since there are a lot of examples in the `doctest` format. And continuously testing them will ensure that these examples work and are up-to-date.

There are some problems with it however, which I am trying to resolve at the moment.

```
» pytest --doctest-modules
================================ test session starts =================================
platform darwin -- Python 3.7.1, pytest-4.2.0, py-1.7.0, pluggy-0.8.1
rootdir: /Users/sobolev/Documents/github/punq, inifile:
plugins: cov-2.6.1
collected 19 items / 2 errors / 17 selected

======================================= ERRORS =======================================
_________________________ ERROR collecting punq/__init__.py __________________________
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:932: in find
    self._find(tests, obj, name, module, source_lines, globs, {})
.venv/lib/python3.7/site-packages/_pytest/doctest.py:403: in _find
    self, tests, obj, name, module, source_lines, globs, seen
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:995: in _find
    globs, seen)
.venv/lib/python3.7/site-packages/_pytest/doctest.py:403: in _find
    self, tests, obj, name, module, source_lines, globs, seen
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:982: in _find
    test = self._get_test(obj, name, module, globs, source_lines)
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:1066: in _get_test
    filename, lineno)
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:668: in get_doctest
    return DocTest(self.get_examples(string, name), globs,
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:682: in get_examples
    return [x for x in self.parse(string, name)
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:644: in parse
    self._parse_example(m, name, lineno)
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:714: in _parse_example
    lineno + len(source_lines))
../../../.pyenv/versions/3.7.1/lib/python3.7/doctest.py:800: in _check_prefix
    (lineno+i+1, name, line))
E   ValueError: line 26 of the docstring for punq.InvalidForwardReferenceException has inconsistent leading whitespace: "  punq.InvalidForwardReferenceException: name 'Dependency' is not defined"
___________________________ ERROR collecting punq/py_36.py ___________________________
punq/py_36.py:1: in <module>
    from typing import List, _ForwardRef
E   ImportError: cannot import name '_ForwardRef' from 'typing' (/Users/sobolev/.pyenv/versions/3.7.1/lib/python3.7/typing.py)
!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!
============================== 2 error in 0.55 seconds ===============================
```

**Comment by bobthemighty** [comment]
That's trying to import the python 36 _ForwardRef support even though it's running under python 3.7. Not sure why that would happen, honestly.

I did try setting up docttest but I became disillusioned. It seems like it's mostly useful when all your use-cases are one-liners.

```
>>> container = Container()
>>> container.register('foo', bar)
```
fails for me with "container is not defined".

**Comment by sobolevn** [comment]
I have made a big progress, now I have new failures. But, it is still WIP. I will create a PR for you to check the direction I am going.

**Comment by bobthemighty** [comment]
Fixed by #14
